File: csc/2019/2.LockFreeSet/KalinaAI/LockFreeSetImplTest.java
import com.devexperts.dxlab.lincheck.LinChecker;
import com.devexperts.dxlab.lincheck.LoggingLevel;
import com.devexperts.dxlab.lincheck.Options;
import com.devexperts.dxlab.lincheck.annotations.Operation;
import com.devexperts.dxlab.lincheck.annotations.Param;
import com.devexperts.dxlab.lincheck.paramgen.IntGen;
import com.devexperts.dxlab.lincheck.strategy.stress.StressCTest;
import com.devexperts.dxlab.lincheck.strategy.stress.StressOptions;
import org.junit.Test;

import java.util.HashSet;
import java.util.Iterator;
import java.util.Set;

@Param(name = "value", gen = IntGen.class, conf = "1:5")
@StressCTest
public class LockFreeSetImplTest {
    private LockFreeSet<Integer> set = new LockFreeSetImpl<>();;

    @Operation
    public Boolean add(@Param(name = "value") int value) {
        return set.add(value);
    }
    
    @Operation
    public Boolean remove(@Param(name = "value") int value) {
        return set.remove(value);
    }

    @Operation
    public Boolean contains(@Param(name = "value") int value) {
        return set.contains(value);
    }

    @Operation
    public Boolean isEmpty() {
        return set.isEmpty();
    }

    @Operation
    public Set<Integer> iterator() {
        Set<Integer> result = new HashSet<>();
        for (Iterator<Integer> it = set.iterator(); it.hasNext(); ) {
            result.add(it.next());
        }
        return result;
    }

    @Test
    public void test() {
        Options opts = new StressOptions()
                .iterations(500)
                .threads(3)
                .actorsPerThread(5)
                .logLevel(LoggingLevel.INFO);
        LinChecker.check(LockFreeSetImplTest.class, opts);
    }
}

File: csc/2019/2.LockFreeSet/ChernigovskayaLA/src/test/kotlin/LockFreeSetImplTest.kt
import com.devexperts.dxlab.lincheck.LinChecker
import com.devexperts.dxlab.lincheck.LoggingLevel.INFO
import com.devexperts.dxlab.lincheck.annotations.Operation
import com.devexperts.dxlab.lincheck.annotations.Param
import com.devexperts.dxlab.lincheck.paramgen.IntGen
import com.devexperts.dxlab.lincheck.strategy.stress.StressCTest
import com.devexperts.dxlab.lincheck.strategy.stress.StressOptions
import org.junit.Test

@Param(name = "key", gen = IntGen::class, conf = "1:5")
@StressCTest
class LockFreeSetImplTest {
    private val set = LockFreeSetImpl<Int>()

    val isEmpty: Boolean
        @Operation
        get() = set.isEmpty()

    @Operation(params = ["key"])
    fun add(x: Int): Boolean {
        return set.add(x)
    }

    @Operation(params = ["key"])
    operator fun contains(x: Int): Boolean {
        return set.contains(x)
    }

    @Operation(params = ["key"])
    fun remove(x: Int): Boolean {
        return set.remove(x)
    }

    @Test
    fun test() {
        val opts = StressOptions()
            .iterations(100)
            .threads(3)
            .actorsPerThread(3)
        LinChecker.check(LockFreeSetImplTest::class.java, opts)
    }
}
